src/System.Diagnostics.TraceSource/tests/CorrelationManagerTests.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Diagnostics.Tracing;
+using System.Collections;
+
+namespace System.Diagnostics.TraceSourceTests
+{
+    public class CorrelationManagerTests
+    {
+        [Fact]
+        public void CorrelationManager_CheckStack()
+        {
+            Trace.CorrelationManager.StartLogicalOperation("one");
+            Trace.CorrelationManager.StartLogicalOperation("two");
+
+            // 2 operations in progress
+            ValidateStack(Trace.CorrelationManager.LogicalOperationStack, "two", "one");
+
+            Trace.CorrelationManager.StopLogicalOperation();
+            // 1 operation in progress
+            ValidateStack(Trace.CorrelationManager.LogicalOperationStack, "one");
+
+            Trace.CorrelationManager.StopLogicalOperation();
+            // 0 operation in progress
+            ValidateStack(Trace.CorrelationManager.LogicalOperationStack, Array.Empty<object>());
+        }
+
+        [Fact]
+        public void CorrelationManager_NullOperationId()
+        {
+            Assert.Throws<ArgumentNullException>("operationId", () => Trace.CorrelationManager.StartLogicalOperation(null));
+        }
+
+        [Fact]
+        public void CorrelationManager_EmptyStack()
+        {
+            Assert.Throws<InvalidOperationException>(() => Trace.CorrelationManager.StopLogicalOperation());
+        }
+
+        [Fact]
+        public void CorrelationManager_ActivityId()
+        {
+            Trace.CorrelationManager.ActivityId = Guid.Empty;
```

Comment:
This is the default value of ActivityId, right?  If the setter was a nop, this test would still pass as written ;)